src/console/Microsoft.Health.Fhir.Ingest.Console.Common/Extensions/EventProcessingExtensions.cs

```diff
@@ -60,6 +61,36 @@ public static IServiceCollection AddEventCheckpointing(this IServiceCollection s
 
         public static IServiceCollection AddResumableEventProcessor(this IServiceCollection services, IConfiguration config)
         {
+            // if assigned partition processor is enabled, then inject it
+            var partitionLockingOptions = new PartitionLockingBackgroundServiceOptions();
+            config.GetSection("PartitionLocking").Bind(partitionLockingOptions);
+            var partitionLockingEnabled = partitionLockingOptions.Enabled;
+
+            if (partitionLockingEnabled)
+            {
+                services.AddSingleton<ICheckpointClient, StorageCheckpointClient>();
```

Comment:
Should we use [TryAddSingleton](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.extensions.servicecollectiondescriptorextensions.tryaddsingleton?view=dotnet-plat-ext-7.0) instead, since we add a `StorageCheckpointClient` in other [places](https://github.com/microsoft/iomt-fhir/blob/main/src/console/Microsoft.Health.Fhir.Ingest.Console.Common/Extensions/EventProcessingExtensions.cs#LL49C10-L49C12)?